var/spack/repos/builtin/packages/comgr/package.py

```diff
@@ -113,6 +114,20 @@ class Comgr(CMakePackage):
     # /opt/rocm, and this breaks the build when /opt/rocm exists.
     patch("hip-tests.patch", when="@:4.2.0")
 
+    def check(self):
+        test_dir = join_path(self.build_directory, "test")
+
+        # find all test executables
+        cmd = "find " + test_dir + " -maxdepth 1 -type f -executable -name '*test'"
```

Comment:
Can you write this in Python?

Reply:
Regarding `run_test,` I used it here because all the rocm components I've seen that currently have spack testing enabled(hipblas, rocblas, rocfft, etc.), all use `run_test` for running the test binaries.

Reply:
I removed the whole `check` function and it now uses ctest for testing.